LowLisp Features.md

```diff
@@ -1,3 +1,37 @@
 # Features Necessary for LowLisp
 
 This document lists everything that will be needed in LowLisp, in order to implement HighLisp.
+
+
+## Language
+
+### Functions
+
+* Are not first class, but experience only a small amount of discrimination. There are function pointers (as in C), and attempting to assign a function to a variable just stores its respective address.
+* Cannot be redefined. However, code can modify a function pointer that will be called in the future. This can be used to implement function redefinition in higher-level languagues.
+* Lack reflection and metaprogramming. Code cannot see which S-expressions make up a function, and cannot modify the function in any way other than overwriting its machine code in memory (probably a bad idea).
+* Must have their parameters and return values defined completely (the generic pointer counts as a defined type).
+
+### Variables
+
+* Are lexically scoped.
+* Are allocated on the stack when declared inside of a function.
+* Are allocated on the heap when declared outside of a function.
+* Must have a type that is known at compile-time, which must make include their size in memory.
+
+### Pointers
+
+* By default are reference-counted, and automatically deallocate their target once the last of their kind falls into oblivion.
+* Are also available in raw form.
+* Must point to a type known at compile time.
```

Comment:
This should mention that the size of that which it points to does not need to be known at compile time, such as with array pointers.

Reply:
Alright. I was actually thinking about writing that the size *must* be known, but I know see that that makes no sense. I'll make sure to add that.